### ISSUE TITLE
[backend dependency] Price is not formatted according to language changes

### ISSUE THREAD

**Issue by hazemfarahat** [SAP/spartacus#8118]
Changing the language in Spartacus does not reflect number formats for prices. See screenshots: the price in German should appear:  $485,57

The i18Service in the backend return the correct locale though, and the price formatting is happening in the backend in the ProductPricePopulator.

I believe it could be an ngrx store issue where the prices are not invalidated after changing the language.

**To Reproduce**
1. Go to PDP or SERP
2. Click on the language selector 
3. Change the language from English to German and back

**Expected behavior**
The price should be formatted as $485,57 in german and $485.57 in English

**Screenshots**
![image](https://user-images.githubusercontent.com/2180376/86469100-08fff880-bd39-11ea-8611-8873d45d6957.png)

**Desktop:**
 - OS: [e.g. iOS]: Windows
 - Browser chrome,
 - Version 83.0.4103.116

**Additional context**
I can reproduce the issue on the reference demo site: https://spartacus.c39j2-walkersde1-d4-public.model-t.cc.commerce.ondemand.com/electronics-spa/de/USD/product/358639/DSC-N1

I can also reproduce it locally on my system which using the latest Spartacus libraries currently 2.0.3

**Comment by kacperknapik** [comment]
To explain in more detail:

**Expected behavior:**
The price should be formatted as $485,57 **(with comma)** in german and $485.57 **(with period)** in English

**Comment by kacperknapik** [comment]
Might be backend-related. Looking into that

**Comment by kacperknapik** [comment]
Backend side currently being checked by backend team. Putting this ticket on hold.

**Comment by kacperknapik** [comment]
blocked by [C3PO-2417](https://cxjira.sap.com/browse/C3PO-2417)

**Comment by Xymmer** [comment]
@hazemfarahat unfortunately this ticket will be set to "future" as we rely on an update to the backend. We'll update this ticket when an improvement is released. Thanks for letting us know about this issue.
